### DISCUSSION TITLE
Making Next.js work with nw.js

### DISCUSSION THREAD

**Discussion by BjoernRave** [vercel/next.js#10712] [Help]
Did anyone ever try to get [nw.js](https://github.com/nwjs/nw.js) running with next.js?

**Comment by jerrygreen** [comment]
Haven't tried it, but I'm also curious (google led me here). Sounds as an interesting opportunity to make complex yet lightweight desktop apps

**Reply by jan-wilhelm** [reply]
@JerryGreen @BjoernRave did you guys ever find a good solution to this?

**Reply by jerrygreen** [reply]
Nope, I ended up using electron anyway

**Reply by raitucarp** [reply]
electron has different approach, I think next.js is more close to nwjs, because of its nature combining nodejs with chrome without ipc

**Reply by snyamson** [reply]
> Nope, I ended up using electron anyway

How did you do that?

**Reply by jerrygreen** [reply]
Thankfully to this package, [nextron](https://github.com/saltyshiomix/nextron), it was really easy to setup NextJS + Electron.

But it was two years ago.

Now I prefer to start new projects using:

[Svelte](https://svelte.dev/) + [vite](https://vitejs.dev/) + [tauri](https://tauri.app/) + [WebView2](https://developer.microsoft.com/en-us/microsoft-edge/webview2) embedded web renderer. It will be harder to setup than just to use Nextron, so I don’t recommend it for newbies people in js community, but in the end you will have like 3-5 megabytes executable. With almost the same desktop capabilities, like placing icon in tray etc.

Svelte has so much nicer-looking code btw. Much better than React. Even though I spent many years writing in React, I have to admit Svelte is just better. But might be a preference though: you can use the same stack with NextJS, or Solid, or whatever. It’s not dependent by UI-library. For nextjsy routes in Svelte there’s [Routify](https://github.com/roxiness/routify) library.

**Comment by ayushmanchhabra** [comment]
https://github.com/nwutils/nw-next-example
The dev setup can be generalised for most [front](https://github.com/nwutils/nw-vue3-boilerplate) [end](https://github.com/nwutils/nw-react-example) frameworks. I have to figure out how/if it can be built using `nw-builder`.

**Reply by LazarPetrovic1** [reply]
Do we know how to add a `next.cofig` in there? It keeps giving me obnoxious errors to no end
